pkg/controller/ibmblockcsi/ibmblockcsi_controller.go

```diff
@@ -493,6 +461,50 @@ func (r *ReconcileIBMBlockCSI) reconcileServiceAccount(instance *ibmblockcsi.IBM
 			if err != nil {
 				return err
 			}
+
+			controllerStatefulset, err := r.getControllerStatefulSet(instance)
+			if err != nil {
+				return err
+			}
+
+			nodeDaemonSet, err := r.getNodeDaemonSet(instance)
+			if err != nil {
+				return err
+			}
+
+			if controllerServiceAccountName == sa.Name {
+				controllerlogger := log.WithValues("Resource Type", "Controller")
+				controllerPod := &corev1.Pod{}
+				err := r.getControllerPod(controllerStatefulset, controllerPod)
+				if err != nil {
+					controllerlogger.Error(err, "failed to get controller pod")
+					return err
+				}
+
+				controllerlogger.Info("controller requires restart",
+							"ReadyReplicas", controllerStatefulset.Status.ReadyReplicas,
+							"Replicas", controllerStatefulset.Status.Replicas)
+				controllerlogger.Info("restarting csi controller")
+				rErr := r.restartControllerPod(controllerPod)
+				
+				if rErr != nil {
+					return rErr
+				}
+			}
+			if nodeServiceAccountName == sa.Name {
+				nodelogger := log.WithValues("Resource Type", "Node DaemonSet")
+				nodelogger.Info("node rollout requires restart",
+								"DesiredNumberScheduled", nodeDaemonSet.Status.DesiredNumberScheduled,				
+								"NumberAvailable", nodeDaemonSet.Status.NumberAvailable)
+				nodelogger.Info("csi node stopped being ready - restarting it")
```

Comment:
why the separate loggers? consider using this function's common logger

Reply:
because the resource is different and in the logs I don't see `node rollout requires restart, resource service account` for example

Reply:
done